File: gaia-sdk-python/gaia_sdk/graphql/request/type/CreatedCodeImpulse.py

from gaia_sdk.graphql.request.type.Code import Code

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry


class CreatedCodeImpulse(list):
    """
    Impulse which indicates the resulf of a create code impulse
    """

    def id(self):
        self.append(lambda x: "id")

    """
    the code instance
    """
    def data(self, config: Callable[['Code'], None]):
        def callback(registry: VariableRegistry):
            entity = Code()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/CreatedCodeImpulse.ts

import {Code} from "./Code";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

/**
 * Impulse which indicates the result of a create code impulse
 */
export class CreatedCodeImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "CreatedCodeImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the code instance
     */
    public data = (config: (_:Code) => void) => this.push((registry) => {
        const entity = new Code();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

